northstar/src/api/client.rs

```diff
@@ -489,7 +484,39 @@ impl<'a> Client {
         {
             Response::Ok(()) => Ok(()),
             Response::Err(e) => Err(Error::Api(e)),
-            _ => Err(Error::Protocol),
+            _ => {
+                self.fuse();
+                Err(Error::Protocol)
+            }
+        }
+    }
+
+    /// Store a notification in the notification queue
+    fn queue_notification(&mut self, notification: Notification) -> Result<(), Error> {
+        if let Some(notifications) = &mut self.notifications {
+            if notifications.len() == notifications.capacity() {
+                self.fuse();
+                Err(Error::LaggedNotifications)
+            } else {
+                notifications.push_back(notification);
+                Ok(())
+            }
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Set the fused flag
+    fn fuse(&mut self) {
+        self.fused = true;
+    }
+
+    /// Return Error::Stopped if the client is fused
+    fn fused(&self) -> Result<(), Error> {
+        if self.fused {
+            Err(Error::Stopped)
+        } else {
+            Ok(())
```

Comment:
using the term `fuse` here is really confusing!

Reply:
It's borrowed from `futures`: https://docs.rs/futures/0.3.15/futures/future/trait.FutureExt.html#method.fuse.